qiskit_ibm/runtime/runtime_job.py

```diff
@@ -188,6 +188,29 @@ def error_message(self) -> Optional[str]:
         self._set_status_and_error_message()
         return self._error_message
 
+    def wait_for_closed_stream(self, timeout: Optional[float] = None, wait: float = 5) -> None:
```

Comment:
This method should be called `wait_for_final_state`. It should replace the current `wait_for_final_state` method. Only difference is instead of polling for status every 5 secs, we should just wait for stream to close and then poll status and return it.